docs/content/manual/manual.yml

```diff
@@ -2924,32 +2944,72 @@ sections:
             input: '10'
             output: ['[0,9,5]']
 
+      - title: "`reduce`"
+        body: |
+
+          The `reduce` syntax allows you to combine all of the results of
+          an expression by accumulating them into a single answer.
+          The form is `reduce EXP as $var (INIT; UPDATE)`.
+          As an example, we'll pass `[1,2,3]` to this expression:
+
+              reduce .[] as $item (0; . + $item)
+
+          For each result that `.[]` produces, `. + $item` is run to
+          accumulate a running total, starting from 0 as the input value.
+          In this example, `.[]` produces the results 1, 2, and 3, so the
+          effect is similar to running something like this:
+
+              0 | (1 as $item | . + $item) |
+                  (2 as $item | . + $item) |
+                  (3 as $item | . + $item)
+
+        examples:
+          - program: 'reduce .[] as $item (0; . + $item)'
+            input: '[1,2,3,4,5]'
+            output: ['15']
+
+          - program: 'reduce .[] as [$i,$j] (0; . + $i * $j)'
+            input: '[[1,2],[3,4],[5,6]]'
+            output: ['44']
+
+          - program: 'reduce .[] as {$x,$y} (null; .x += $x | .y += [$y])'
+            input: '[{"x":"a","y":1},{"x":"b","y":2},{"x":"c","y":3}]'
+            output: ['{"x":"abc","y":[1,2,3]}']
+
       - title: "`foreach`"
         body: |
 
           The `foreach` syntax is similar to `reduce`, but intended to
           allow the construction of `limit` and reducers that produce
-          intermediate results (see example).
+          intermediate results.
 
           The form is `foreach EXP as $var (INIT; UPDATE; EXTRACT)`.
-          Like `reduce`, `INIT` is evaluated once to produce a state
-          value, then each output of `EXP` is bound to `$var`, `UPDATE`
-          is evaluated for each output of `EXP` with the current state
-          and with `$var` visible.  Each value output by `UPDATE`
-          replaces the previous state.  Finally, `EXTRACT` is evaluated
-          for each new state to extract an output of `foreach`.
+          As an example, we'll pass `[1,2,3]` to this expression:
+
+              foreach .[] as $item (0; . + $item; . * 2)
```

Comment:
Possibly show "expansion" of foreach similar to for reduce?

Reply:
Curious what it looks like. Following query is the first one I come up with.
```jq
0 | (1 as $item | . + $item) | (. * 2), (
    (2 as $item | . + $item) | (. * 2), (
    (3 as $item | . + $item) | (. * 2)))
```

Reply:
Looks correct 👍  i remember seeing a comment in the jaq code explaining this in a similar way https://github.com/01mf02/jaq/blob/bac33c3a1747ea390de368ebc64e9175b5218ce5/jaq-core/src/filter.rs#L40-L62

Reply:
But think i it's probably easier to follow with a concrete example like for reduce in the jq docs?